flink-runtime/src/main/java/org/apache/flink/runtime/deployment/TaskDeploymentDescriptor.java

```diff
@@ -141,9 +146,16 @@ public TaskDeploymentDescriptor(
 			List<BlobKey> requiredJarFiles, List<URL> requiredClasspaths,
 			int targetSlotNumber) {
 
-		this(appId, jobID, vertexID, executionId, taskName, indexInSubtaskGroup, numberOfSubtasks, attemptNumber,
-				jobConfiguration, taskConfiguration, invokableClassName, producedPartitions,
-				inputGates, requiredJarFiles, requiredClasspaths, targetSlotNumber, null, -1);
+		this(appId, jobID, vertexID, executionId, new ExecutionConfig(), taskName, indexInSubtaskGroup,
```

Comment:
why is there a constructor variant of the TDD that creates an empty execution config?